tools/build_config/Jenkinsfile

```diff
@@ -163,11 +187,16 @@ pipeline {
     stage('Notify') {
       steps {
         script {
-          if (env.BRANCH_NAME) {
-            currentBuild.description = "Branch: ${env.BRANCH_NAME}"
-          } else if (env.PR_NUMBER) {
-            currentBuild.displayName = "#${env.BUILD_NUMBER} PR-${env.PR_NUMBER}"
-            currentBuild.description = "Git-SHA: ${env.PR_COMMIT_SHA.take(7)}"
+          // Archive a file containing the git revision being built. This
```

Comment:
For consistency the archive action it should happen as part of `post` section below where the rest of the `archiveArtifacts` directives are placed. The creation of the file isn't a notify action either; that should move either to the point the file is consumed or outside this block

Reply:
The `write_git_revision...` call cannot be made outside a `stage { step { } }` block - since it's using `sh`/`powershell` the job will need to have been allocated a node before it can be called. I could create a new stage? I guess I justified putting it into the "Notify" step, since the purpose of writing the file is to "notify" the Deploy pipeline of what revision was built - I realize that's a bit of stretch...

I'll move the `archiveArtifcats` call